modules/features2d/include/opencv2/features2d.hpp

```diff
@@ -1314,6 +1314,13 @@ CV_EXPORTS_W void drawMatches( InputArray img1, const std::vector<KeyPoint>& key
                              const std::vector<char>& matchesMask=std::vector<char>(), int flags=DrawMatchesFlags::DEFAULT );
 
 /** @overload */
+CV_EXPORTS_W void drawMatches( InputArray img1, const std::vector<KeyPoint>& keypoints1,
+                             InputArray img2, const std::vector<KeyPoint>& keypoints2,
+                             const std::vector<DMatch>& matches1to2, InputOutputArray outImg,
+                             const int matchesThickness, const Scalar& matchColor=Scalar::all(-1),
+                             const Scalar& singlePointColor=Scalar::all(-1), const std::vector<char>& matchesMask=std::vector<char>(),
+                             int flags=DrawMatchesFlags::DEFAULT );
```

Comment:
@vpisarev I believe we should define strategy about changes which should be migrated to OpenCV 5.0 major release

Reply:
Regarding Python bindings: I've tested the code provided by the author of #20095 and it's working fine.

Regarding the documentation: the new parameter should be documented when it's only used in an overloaded function? If so, I can take a look at it.
My first try was to include it as a parameter of the original function, but some errors happened due to the function names inside of binary files, that's why I've created a new function.